### ISSUE TITLE
About simulating data

### ISSUE THREAD

**Issue by shanhaidexiamo** [BUTSpeechFIT/vae_dolphin#1]
Hi ,I want to ask if there only need to simulate tt data? tr and cv data just use 2speaker_reverb? Thanks

**Comment by zmolikova** [comment]
Hi,
yes, the training is done on data without any noise, so just directly 2speaker_reverb. The `create_json.py` step will create `tr_1spk.json` with the corresponding paths.

**Comment by shanhaidexiamo** [comment]
Hello ,thanks for your reply for my question. Recently I try to run your code, and in inferencing stage, I found that the avg sdr was very low in infer_vae_noise_-5-0, the avd sdr was about 0.75 using 3000 tt dataset. so , I want to ask you when you do inference , the 3000 utt are all used or just use part of the tt data such as 1/600 ? By the way , whether the config in your code is the VAE run5 config？ Thank you !

**Comment by zmolikova** [comment]
Hi,
thanks for the interest and for running through the code!
The reported results are on all 3000 utterances in tt dataset. And for all five runs, the configuration is the same, there is just some randomness in the training.
Please note, that the results reported in the tables are improvements with respect to the mixture SDR. But even then, the absolute SDR for the -5-0 case should be more like 1.0. Are your results with the pre-trained model or you trained the VAE from scratch? If from scratch, could you share the Tensorboard logs? It should be in `exp/vae/runs`.
Thank you!

**Comment by shanhaidexiamo** [comment]
Hi，
thanks for your reply , I trained VAE from scratch. And I also open the Tensorboard logs, here is my logs:

![image](https://user-images.githubusercontent.com/48786488/105833907-cff4e180-6004-11eb-9783-27f849be284a.png)
![image](https://user-images.githubusercontent.com/48786488/105834004-ed29b000-6004-11eb-8ad7-3871204de024.png)
![image](https://user-images.githubusercontent.com/48786488/105834054-f7e44500-6004-11eb-943a-bf8b1b83705f.png)
![image](https://user-images.githubusercontent.com/48786488/105834091-029eda00-6005-11eb-80f0-f7716f259c4c.png)
![image](https://user-images.githubusercontent.com/48786488/105834131-0cc0d880-6005-11eb-8d9e-da948ccbfa2f.png)

Is there anything wrong ? Thank you!

**Comment by zmolikova** [comment]
The loss functions look just right. I realized the problem is elsewhere. When I compute the SDR results, I select the best permutation of the sources based on SDR values rather than SIR values, as it is done in `mir_eval` and `pb_bss`. I forgot to reflect this in the repository. I fixed this now (commit https://github.com/BUTSpeechFIT/vae_dolphin/commit/de5956b3bb6cb48428997d499ab0efb8b0aa2b91). Please, try to rerun the scoring with `compute_sdr.py`.

Let me also comment a bit on why I choose to evaluate in this way. When I did the evaluation selecting the permutation with SIR, I noticed that for some recordings the permutations were wrong (wrong considering SDR, but also subjectively wrong when I listened to the outputs). This caused some very low SDR values for utterances that were reasonably separated. For this reason, I changed the evaluation to select the permutation using SDR.

**Comment by shanhaidexiamo** [comment]
Thanks a lot for your patient answer. I try to run it today, the avg sdr in -5-0 comes to 1.6dB, better than before, but still not enough. So I'm going to rerun your code, including the data simulation from the beginning, and I'll give you feedback in time again. Thanks again!
![image](https://user-images.githubusercontent.com/48786488/106015402-6a365180-60f9-11eb-984f-35e64cfa0491.png)

**Comment by zmolikova** [comment]
The result on the output of `compute_sdr.py` is the absolute SDR, while the numbers reported in the tables are relative improvements with respect to the mixture SDR. To get the improvement, you should thus compute a difference between this number and mixture SDR (from here https://github.com/BUTSpeechFIT/vae_dolphin/blob/main/RESULTS_VAE.md or the [paper](https://arxiv.org/abs/2011.11984)). This is 1.59 - (-6.89) =  8.48. Even when using the oracle ideal binary mask, the absolute SDR is about 3.2 dB. So your result is in line with the results reported in the paper.

I should make this information more clear in the README and the scoring script.

**Comment by shanhaidexiamo** [comment]
Oops! Because I am new at this and ignore this point.Thank you very much for answering my doubts!

**Comment by zmolikova** [comment]
Thank you for helping me spot the issues!
